File: src/main/java/com/zerone/math/Mathf.java
package com.zerone.math;

public final class Mathf {

    public static final float E = 2.718281828459045f;
    public static final float PI = 3.141592653589793f;

    public static final float DEG_TO_RADIANS = PI / 180.0f;
    public static final float RAD_TO_DEGREES = 180.0f / PI;

    public static float sin(float var0) {
        return (float) Math.sin(var0);
    }

    public static float cos(float var0) {
        return (float) Math.cos(var0);
    }

    public static float tan(float var0) {
        return (float) Math.tan(var0);
    }

    public static float asin(float var0) {
        return (float) Math.asin(var0);
    }

    public static float acos(float var0) {
        return (float) Math.acos(var0);
    }

    public static float atan(float var0) {
        return (float) Math.atan(var0);
    }

    public static float toRadians(float angdeg) {
        return angdeg * DEG_TO_RADIANS;
    }

    public static float toDegrees(float angrad) {
        return angrad * RAD_TO_DEGREES;
    }

    public static float exp(float var0) {
        return (float) Math.exp(var0);
    }

    public static float log(float var0) {
        return (float) Math.log(var0);
    }

    public static float log10(float var0) {
        return (float) Math.log10(var0);
    }

    public static float sqrt(float var0) {
        return (float) Math.sqrt(var0);
    }

    public static float cbrt(float var0) {
        return (float) Math.cbrt(var0);
    }

    public static float IEEEremainder(float var0, float var2) {
        return (float) Math.IEEEremainder(var0, var2);
    }

    public static float ceil(float var0) {
        return (float) Math.ceil(var0);
    }

    public static float floor(float var0) {
        return (float) Math.floor(var0);
    }

    public static float rint(float var0) {
        return (float) Math.rint(var0);
    }

    public static float atan2(float var0, float var2) {
        return (float) Math.atan2(var0, var2);
    }

    public static float pow(float var0, float var2) {
        return (float) Math.pow(var0, var2);
    }

    public static int round(float a) {
        return Math.round(a);
    }

    public static float random() {
        return (float) Math.random();
    }

    public static float abs(float a) {
        return Math.abs(a);
    }

    public static float max(float a, float b) {
        return Math.max(a, b);
    }

    public static float min(float a, float b) {
        return Math.min(a, b);
    }

    public static float ulp(float f) {
        return Math.ulp(f);
    }

    public static float signum(float f) {
        return Math.signum(f);
    }

    public static float sinh(float var0) {
        return (float) Math.sinh(var0);
    }

    public static float cosh(float var0) {
        return (float) Math.cosh(var0);
    }

    public static float tanh(float var0) {
        return (float) Math.tanh(var0);
    }

    public static float hypot(float var0, float var2) {
        return (float) Math.hypot(var0, var2);
    }

    public static float expm1(float var0) {
        return (float) Math.expm1(var0);
    }

    public static float log1p(float var0) {
        return (float) Math.log1p(var0);
    }

    public static float copySign(float magnitude, float sign) {
        return Math.copySign(magnitude, sign);
    }

    public static int getExponent(float f) {
        return Math.getExponent(f);
    }

    public static float scalb(float f, int scaleFactor) {
        return Math.scalb(f, scaleFactor);
    }
}

File: src/main/kotlin/com/zerone/math/Mathf.kt
package com.zerone.math

class Mathf {

    companion object {
        const val E = 2.718281828459045f
        const val PI = 3.141592653589793f

        const val DEG_TO_RADIANS = PI / 180.0f
        const val RAD_TO_DEGREES = 180.0f / PI

        @JvmStatic fun sin(var0: Float): Float {
            return Math.sin(var0.toDouble()).toFloat()
        }

        @JvmStatic fun cos(var0: Float): Float {
            return Math.cos(var0.toDouble()).toFloat()
        }

        @JvmStatic fun tan(var0: Float): Float {
            return Math.tan(var0.toDouble()).toFloat()
        }

        @JvmStatic fun asin(var0: Float): Float {
            return Math.asin(var0.toDouble()).toFloat()
        }

        @JvmStatic fun acos(var0: Float): Float {
            return Math.acos(var0.toDouble()).toFloat()
        }

        @JvmStatic fun atan(var0: Float): Float {
            return Math.atan(var0.toDouble()).toFloat()
        }

        @JvmStatic fun toRadians(angdeg: Float): Float {
            return angdeg * DEG_TO_RADIANS
        }

        @JvmStatic fun toDegrees(angrad: Float): Float {
            return angrad * RAD_TO_DEGREES
        }

        @JvmStatic fun exp(var0: Float): Float {
            return Math.exp(var0.toDouble()).toFloat()
        }

        @JvmStatic fun log(var0: Float): Float {
            return Math.log(var0.toDouble()).toFloat()
        }

        @JvmStatic fun log10(var0: Float): Float {
            return Math.log10(var0.toDouble()).toFloat()
        }

        @JvmStatic fun sqrt(var0: Float): Float {
            return Math.sqrt(var0.toDouble()).toFloat()
        }

        @JvmStatic fun cbrt(var0: Float): Float {
            return Math.cbrt(var0.toDouble()).toFloat()
        }

        @JvmStatic fun IEEEremainder(var0: Float, var2: Float): Float {
            return Math.IEEEremainder(var0.toDouble(), var2.toDouble()).toFloat()
        }

        @JvmStatic fun ceil(var0: Float): Float {
            return Math.ceil(var0.toDouble()).toFloat()
        }

        @JvmStatic fun floor(var0: Float): Float {
            return Math.floor(var0.toDouble()).toFloat()
        }

        @JvmStatic fun rint(var0: Float): Float {
            return Math.rint(var0.toDouble()).toFloat()
        }

        @JvmStatic fun atan2(var0: Float, var2: Float): Float {
            return Math.atan2(var0.toDouble(), var2.toDouble()).toFloat()
        }

        @JvmStatic fun pow(var0: Float, var2: Float): Float {
            return Math.pow(var0.toDouble(), var2.toDouble()).toFloat()
        }

        @JvmStatic fun round(a: Float): Int {
            return Math.round(a)
        }

        @JvmStatic fun random(): Float {
            return Math.random().toFloat()
        }

        @JvmStatic fun abs(a: Float): Float {
            return Math.abs(a)
        }

        @JvmStatic fun max(a: Float, b: Float): Float {
            return Math.max(a, b)
        }

        @JvmStatic fun min(a: Float, b: Float): Float {
            return Math.min(a, b)
        }

        @JvmStatic fun ulp(f: Float): Float {
            return Math.ulp(f)
        }

        @JvmStatic fun signum(f: Float): Float {
            return Math.signum(f)
        }

        @JvmStatic fun sinh(var0: Float): Float {
            return Math.sinh(var0.toDouble()).toFloat()
        }

        @JvmStatic fun cosh(var0: Float): Float {
            return Math.cosh(var0.toDouble()).toFloat()
        }

        @JvmStatic fun tanh(var0: Float): Float {
            return Math.tanh(var0.toDouble()).toFloat()
        }

        @JvmStatic fun hypot(var0: Float, var2: Float): Float {
            return Math.hypot(var0.toDouble(), var2.toDouble()).toFloat()
        }

        @JvmStatic fun expm1(var0: Float): Float {
            return Math.expm1(var0.toDouble()).toFloat()
        }

        @JvmStatic fun log1p(var0: Float): Float {
            return Math.log1p(var0.toDouble()).toFloat()
        }

        @JvmStatic fun copySign(magnitude: Float, sign: Float): Float {
            return Math.copySign(magnitude, sign)
        }

        @JvmStatic fun getExponent(f: Float): Int {
            return Math.getExponent(f)
        }

        @JvmStatic fun scalb(f: Float, scaleFactor: Int): Float {
            return Math.scalb(f, scaleFactor)
        }
    }
}
